accepted/0000-caasp-nodes.md

```diff
@@ -0,0 +1,274 @@
+    - The corresponding action must fail if scheduled via XMLRPC: `SystemHandler#scheduleReboot`
+- Package upgrade:
+  - If the targeted minion is a `caasp_node` system type, the action for upgrade a package:
+    - Must be hidden from the UI and the corresponding action must fail: code paths `/systems/details/packages/UpgradableList`, `systems/ssm/PackageUpgrade`, `ActionChainManager#schedulePackageUpgrade`, `ActionChainManager#schedulePackageUpgrades`
+    - The corresponding action must fail if scheduled via XMLRPC: [`SystemHandler#schedulePackageInstall`]
+- Package removal:
+  - If the targeted minion is a `caasp_node` system type, the action for removing a package:
+    - Must be hidden from the UI and the corresponding action must fail: `/systems/details/packages/RemoveConfirm`, `systems/ssm/PackageRemove`, `ActionChainManager#schedulePackageRemoval`, `ActionChainManager#schedulePackageRemoval`
+    - The corresponding action must fail if scheduled via XMLRPC: `SystemHandler#schedulePackageRemove`
+- Patch apply:
+  - If the targeted minion is a `caasp_node` system type, the action for applying a patch:
+    - Must be hidden from the UI and the corresponding action must fail: `/systems/details/ErrataList.do`, `/systems/ssm/ListErrata`, `ErrataManager#applyErrata`
+    - The corresponding action must fail if scheduled via XMLRPC: [`SystemHandler#scheduleApplyErrata`]
+
+Value of this solution:
+* This workflow does not constitute a radical change for a user
+* Change does not involve Salt, the user can still break its cluster by issuing Salt commands
+* Requires changes in Uyuni/SUSE Manager core code
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+<!-- Why should we **not** do this?
+
+  * obscure corner cases
+  * will it impact performance?
+  * what other parts of the product will be affected?
+  * will the solution be hard to maintain in the future? -->
+
+The user can craft a Salt state that executes one of the forbidden operations and breaks the cluster. This possibility is outside the scope of the RFC.
+
+# Alternatives
+[alternatives]: #alternatives
+
+<!-- - What other designs/options have been considered?
+- What is the impact of not doing this? -->
+
+## Future work
+
+In the future, we can team up with cluster providers and ask to `skuba` and other [cluster provider managers](https://trello.com/c/2X01ypO4/10-cluster-awarenesss-workshop-suse-manager-caasp-ha-cap-ses#comment-5dc1bc4567c6031ae57bcb04) to implement all these forbidden actions that require special handling. In that case, Uyuni/SUSE Manager can remove all the specific features described in this RFC and offload the task to the cluster provider manager.
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+
+* Deciding which action locking mechanism to take: Salt minion blackout or Uyuni/SUSE Manager action locking
```

Comment:
I think both are needed. If only the actions are locked in the UI/API the user still has the option to use the Salt command line or the Remote Command UI to execute any of the forbidden actions.

Reply:
If we go with Salt minion blackout then we do not need Uyuni/SUSE Manager action locking (unless we want to forbid the forbidden actions when blackout is disabled).
With minion blackout, every action scheduled by Uyuni/SUSE Manager will fail - at Salt level the action will not be executed (even `cmd.run` if not whitelisted).

Reply:
> With minion blackout, every action scheduled by Uyuni/SUSE Manager will fail - at Salt level the action will not be executed (even `cmd.run` if not whitelisted).

That would do the trick for the first iteration but I think it's not very user friendly. IMHO users should see clearly that a minion is locked for certain operations.

Reply:
From a UI and from a [perf POV](https://github.com/uyuni-project/uyuni-rfc/pull/26#discussion_r345249600) I agree. Pushing a correction in this.